src/Header.php

```diff
@@ -0,0 +1,35 @@
+<?php namespace DeSmart\Mailer;
+
+class Header
+{
+    /** @var string */
+    protected $name;
+    /** @var string */
+    protected $value;
+
+    /**
+     * @param string $name
+     * @param string $value
+     */
+    public function __construct($name, $value)
+    {
+        $this->name = $name;
```

Comment:
When it comes to headers, Mandrill and SendGrid accepts most of smtp headers and there is quite lot of them, so I think validation of headers is not feasible.